core/src/test/java/kafka/testkit/KafkaClusterTestKitTest.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka.testkit;
+
+import org.hamcrest.Matcher;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import java.nio.file.Paths;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class KafkaClusterTestKitTest {
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    public void testCreateClusterAndCloseWithMultipleLogDirs(boolean combined) {
+        try (KafkaClusterTestKit cluster = new KafkaClusterTestKit.Builder(
+                new TestKitNodes.Builder().
+                        setBrokerNodes(5, 2).
+                        setCombined(combined).
+                        setNumControllerNodes(3).build()).build()) {
+
+            assertEquals(5, cluster.nodes().brokerNodes().size());
+            assertEquals(3, cluster.nodes().controllerNodes().size());
+
+            cluster.nodes().brokerNodes().forEach((brokerId, node) -> {
+                assertEquals(2, node.logDataDirectories().size());
+                Matcher<Iterable<? extends String>> matcher = containsInAnyOrder(String.format("broker_%d_data0", brokerId), String.format("broker_%d_data1", brokerId));
```

Comment:
It seems we don't use `hamcrest` in core module. Maybe we can put expected/actual folders into a `HashSet`, and then call `assertEqual` to compare them?

Reply:
> is there any reason not to use hamcrest in core when we do use it in other modules?

I'm curious. Maybe our modules have different tastes of code style :smile: